### ISSUE TITLE
Strange behavior deserializing strings into string arrays

### ISSUE THREAD

**Issue by rudism** [ServiceStack/Issues#630]
When deserializing from plain string into a `string[]`, it looks like it's splitting on the space character (which is reasonable behavior), but also dropping the first letter of each resulting element (except for the first element), as demonstrated here using the mono REPL:

```
Enter statements below.
csharp> using ServiceStack;
csharp> "this is a string with a bunch of words".FromJson<string[]>();
{ "this", "s", "string", "ith", "bunch", "f", "ords" }
```

We recently had a bug in our application in which the back-end API was expecting an array where the client was sending a plain string, and we didn't realize for some time that characters were being lost. I think it would make more sense in this situation to throw an exception, or at least split on whitespace without dropping characters from the original string.

**Comment by mythz** [comment]
ServiceStack.Text does its best to deserialize without Exception by design, the error here is that you're trying to deserialize a `string` into a `string[]`.

I've changed it to not lose chars when parsing into a string[] in [this commit](https://github.com/ServiceStack/ServiceStack.Text/commit/f71d08367b0ccc91760ab1cf9b47f98d5254c047)

**Comment by mythz** [comment]
FYI it will also throw an Exception if you enable:

```csharp
Env.StrictMode = true;
```

This change is available from v5.1.1 that's now [available on MyGet](http://docs.servicestack.net/myget).
